### PR TITLE
Done

### PR DESCRIPTION
**Pull request by johnram528**

### CODE REVIEW

File: bin/tictactoe [hunk 1]
```diff
@@ -1,3 +1,34 @@
...
+    player_1 = Player::Human.new("X")
+    player_2 = Player::Human.new("O")
+  when "0"
+    player_1 = Player::Computer.new("X")
+    player_2 = Player::Computer.new("O")
+  else
+    puts "Sorry, try again."
+end
+
+Game.new(player_1, player_2).play
```
Comment (randallreedjr) [inline bin/tictactoe]: Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+class Board
+  attr_accessor :cells, :position
+
+  @cells = []
+  def initialize
+    @cells = Array.new(9, " ")
```
Comment (randallreedjr) [inline lib/board.rb]: I would just call `self.reset!` here and delegate to that method to create the default empty board.

File: lib/board.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  end
+
+  def reset!
+    self.cells.clear
```
Comment (randallreedjr) [inline lib/board.rb]: No need to call `clear` as the contents of `cells` will be overwritten by the assignment.

File: lib/board.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    border = "-----------"
+    puts top_row, border, middle_row, border, bottom_row
+  end
+
+  def position(num)
+    self.cells[num.to_i-1]
+  end
+
+  def turn_count
+    count = self.cells.count {|cell| cell == "X" || cell == "O"}
```
Comment (randallreedjr) [inline lib/board.rb]: You don't need to use an intermediate assignment here. You can just return `self.cells.count {|cell| cell == "X" || cell == "O"}`.

File: lib/players/human.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+class Player::Human < Player
+
+  def move(board)
+    puts "Where do you want to move next?(1-9)"
+    input = gets.strip
```
Comment (randallreedjr) [inline lib/players/human.rb]: You don't need to store your return value in a local variable; just utilize the implicit return value of the expression.
```
def move(board)
  puts "Where would you like to move?"
  gets.chomp
end
```

File: lib/board.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    count
+  end
+
+  def full?
+    if self.turn_count == 9
+      true
```
Comment (randallreedjr) [inline lib/board.rb]: Avoid returning boolean literals from `if` statements, relying instead on the implicit return value of the boolean expression.
```
def full?
  self.turn_count == 9
end
```

File: lib/game.rb [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+     WIN_COMBINATIONS.detect do |combinations|
+       if combinations.all? {|position| board.cells[position] == "X"}
+        @winner = "X"
+        return combinations
+       elsif combinations.all? {|position| board.cells[position] == "O"}
+        @winner = "O"
+        return combinations
+       else
+        @winner = nil
+        false
```
Comment (randallreedjr) [inline lib/game.rb]: In this case, returning the result of `@winner = nil` will be falsey, so `false` is redundant.

**Comment by randallreedjr** [comment]
@johnram528 Great job! I left you a few notes, feel free to respond with any questions or comments. Sorry it took me so long to provide you with feedback, hopefully it’s still useful to you! Closing this PR for now.

**Comment by johnram528** [comment]
Very helpful, thanks!

**Pull request closed without merging**